### ISSUE TITLE
Draft Order API Percentage Discount issue

### ISSUE THREAD

**Issue by piyushghevariya** [Shopify/shopify_api#654]
Hey
I found that In some cases Percentage discount is not working with Draft Order API. It gives me **{:"line_items[0].applied_discount.amount"=>["must correspond to that calculated from the value"]}** error. reference: https://prnt.sc/qdeqm7

Shopify Draft Order API mention in the document to use floor function to apply percentage discount. reference: https://prnt.sc/qdeqm7
I think the issue is with the rails floor function fraction issues also used some other function to get the exact discount value. Can you suggest to me how I can apply floor function in rails?

I have used the following codes for percentage calculation:
`amount = (price * quantity * value).floor / 100`

Here are some examples in which I found this error:
1. 
price: 17.90
quantity: 2
percentage_value: 25
after_discount_price: 8.94
Error: https://prnt.sc/qdeqm7
In this scenario, if the after discount price is 8.95 then it works. But using floor function it can't be accomplished.

2. 
price: 850
quantity: 5
percentage_value: 5
after_discount_price: 212.5
Error: https://prnt.sc/qdexb5
In this scenario all is fine then also it gives the error.

Can you please provide me where I am going wrong or is there any issue from draft order API?
Let me know If you have any questions regarding my query and hope you provide me suggestion/solution to resolve this.

Thanks!!!

**Comment by piyushghevariya** [comment]
Hey
Any update on the above issue?
I need Quick help as we are facing this issue.

Thanks

**Comment by piyushghevariya** [comment]
Hey there, 

Can anyone please provide us the solution? We need quick help on this. 

Any help will be appreciated.

Thanks.

**Comment by tanema** [comment]
I don't think you need to supply that after_discount_price value, Shopify will calculate it for you. Just leave it out.
